src/coreclr/vm/threads.cpp

```diff
@@ -370,7 +383,27 @@ void SetThread(Thread* t)
 #endif
 
     // Clear or set the app domain to the one domain based on if the thread is being nulled out or set
-    t_CurrentThreadInfo.m_pAppDomain = t == NULL ? NULL : AppDomain::GetCurrentDomain();
+    if (t != NULL)
+    {
+        t_CurrentThreadInfo.m_pAppDomain = AppDomain::GetCurrentDomain();
+#if defined(TARGET_UNIX) && !defined(TARGET_WASM)
+        if (!InsertThreadIntoAsyncSafeMap(t->GetOSThreadId64(), t))
+        {
+            // TODO: can we handle this OOM more gracefully?
```

Comment:
We can - if it is one of our threads. However, I do not think it is worth it. The allocation of the thread statics above can crash due to OOM too and I do not think there is a good way to recover from it either. Recovering from corner-case OOMs is just not a thing on Unix.